misc.go

```diff
@@ -0,0 +1,210 @@
+/*
+ * File: misc.go
+ *
+ * Description: Contains a number of helpful util functions.
+ *
+ */
+
+package main
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strings"
+	"time"
+)
+
+//! Wrapper to make golang error funcs seem more C-like
+/*
+ * @param     string    ASCII to dump to stdout
+ *
+ * @return    error     error message, if any
+ */
+func errorf(ascii string) error {
+
+	// Input validation
+	if len(ascii) < 1 {
+		return nil
+	}
+
+	// Attempt to print the content to stdout
+	return fmt.Errorf(ascii)
+}
+
+//! Wrapper to make golang print funcs seem more C-like
+/*
+ * @param     string    ASCII to dump to stdout
+ *
+ * @return    none
+ */
+func printf(ascii string) {
+
+	// Input validation
+	if len(ascii) < 1 {
+		return
+	}
+
+	// Attempt to print the content to stdout
+	fmt.Printf(ascii + "\n")
+}
+
+//! Wrapper to give a log-like appearance to stdout
+/*
+ * @param     string    ASCII to dump to stdout
+ *
+ * @return    none
+ */
+func stdlog(ascii string) {
+
+	// Input validation
+	if len(ascii) < 1 {
+		return
+	}
+
+	// Grab the current time in seconds from epoch.
+	currentTime := time.Now().String()
+
+	// Append the timestamp to the string message.
+	printf("[" + currentTime + "] " + ascii)
+}
+
+//! Function to print out debug messages
+/*
+ * @param     string    ASCII to dump to stdout
+ *
+ * @return    none
+ */
+func debugf(ascii string) {
+
+	// Input validation
+	if len(ascii) < 1 {
+		return
+	}
+
+	// ensure debug mode is actually on
+	if !debugMode {
+		return
+	}
+
+	// Grab the current time in seconds from epoch.
+	currentTime := time.Now().String()
+
+	// Append the timestamp to the string message.
+	printf("[" + currentTime + "] DEBUG - " + ascii)
+}
+
+//! Determine if a given directory location actually exists.
+/*
+ * @param     string    location of the form: /path/to/directory/
+ *
+ * @return    bool      whether or not this location is a POSIX dir
+ */
+func directoryExists(path string) bool {
+
+	// input validation
+	if len(path) < 1 {
+		return false
+	}
+
+	// attempt to read the directory and its contents, if any
+	_, err := ioutil.ReadDir(path)
+
+	// if an error occurred, then assume this is probably not a directory
+	if err != nil {
+		return false
+	}
+
+	// otherwise this succeeded, in which case return true
+	return true
+}
+
+//! Function to grab the hostname at a given context
+/*
+ * @return    string    hostname as a string
+ */
+func getHostname() string {
+
+	// attempt to grab the hostname
+	hostname, err := os.Hostname()
+
+	// safety check, ensure no error occurred
+	if err != nil {
+		return "n/a"
```

Comment:
While I surmise this will almost never fail here, I agree with you, might as well handle errors via the golang philosophy.

https://github.com/rbisewski/lclusterd/commit/555e13bc723cebb2b37ab65f43ceff5bef5160bb